### DISCUSSION TITLE
Loading React scripts as "defer"

### DISCUSSION THREAD

**Discussion by AlexxEG** [microsoft/Dynamics365Commerce.Solutions#202] [Q&A] [answered]
Could the React scripts be loaded with "defer" attributes to shave off a few ms on loading time? Since server rendering is used I believe it should have a positive impact on the initial load.

**Comment by madyke** [comment]
Hi @AlexxEG - I passed this onto our team's React SME, who informed me that we load extension dependencies as modules with a library called SystemJS. SystemJS already [loads modules with a "deferred execution" mode](https://github.com/systemjs/systemjs/blob/main/docs/system-register.md#deferred-execution), so extension dependencies should already be loaded with "defer" by default.

**Comment by AlexxEG** [comment]
Hey @madyke - I was referring to the "react.production.min.js" and "react-dom.production.min.js" scripts that are at the bottom of the body. Could it be beneficial to move these to the head and add the "defer" attribute to them instead so that they're loaded async while the HTML is parsed?

Unless of course there is a specific reason it's done the way it is now.

**Reply by madyke** [reply]
Hi @AlexxEG - what product are you referring to? Is this for an E-commerce site? None of the Store commerce or HWS applications are using those scripts as far as I'm aware

**Reply by AlexxEG** [reply]
Yeah correct, I was directed here to create a discussion thread from support to get input on this change. Maybe this is the wrong place for that?

**Reply by madyke** [reply]
Hi @AlexxEG - e-commerce questions are fine here. Store Commerce is going through a React upgrade now, so without any context specified, I assumed it was related to that.

If this is e-commerce related, then @bradyd or someone on his team should be able to review your suggestion.

**Reply by AlexxEG** [reply]
@madyke Got it, thank you!

**Comment by bradyd** [comment, marked as answer]
Hey @AlexxEG, 
We are looking at using defer for the react scripts in the near future. We just need to make sure it doesn't cause any other interesting issues when we enable it.  Currently put the react scripts at the bottom of the body to let the render happen before loading the script.

Brady

**Reply by AlexxEG** [reply]
Hey @bradyd 
Ok sounds good, thanks for answering!
